### PR TITLE
feat: Capture blacklisted specs inside archive

### PR DESCRIPTION
**Pull request by ryan-blakley**
Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:
Capture the name of blacklisted specs, and dump them to a file in the archive. Added parser and spec to parse the blacklisted specs and use in rules.

### CODE REVIEW

File: insights/collect.py [hunk 1]
```diff
@@ -473,6 +480,40 @@ def _parse_broker_exceptions(broker, exceptions_to_report):
...
+    Write out the blacklisted specs to blacklisted_specs.txt, and create
+    a meta-data file for this file. That way it can be loaded when the
+    archive is processed.
+
+    Args:
+        broker (Broker): Broker object used for core collection.
+        output_path (str): Path of the output directory.
+    """
+    with open(os.path.join(output_path, "blacklisted_specs.txt"), "w") as of:
+        of.write(", ".join(broker.blacklisted_specs))
```
Comment (bfahr) [inline insights/collect.py:494]: @ryan-blakley what if you converted this to JSON instead of creating a string here that must be re-parsed?

Comment (ryan-blakley) [reply insights/collect.py]: @bfahr I updated it to output json, and got it working on both core collection and non-core collection and confirmed that the specs and parser works from within a rule.

**Review by bfahr — CHANGES_REQUESTED** [review]

File: insights/parsers/blacklisted.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+"""
```
[the remark this answers is not in the document]
Comment (ryan-blakley) [reply insights/parsers/blacklisted.py:1]: Good catch, fixed!

**Comment by ryan-blakley** [comment]
@subpop When you get a chance if you could review the client code changes in this PR please.

**Comment by ryan-blakley** [comment]
Awesome thank you for reviewing it.

**Comment by ryan-blakley** [comment]
@kgrant-rh @jsvob Since y'alls team was the one who brought up the issue about not being able to tell if spec was blacklisted or not. When y'all have time if you could please review this PR, and let me know if these changes will help with that.

**Pull request merged**